generic_deployment.rst

```diff
@@ -0,0 +1,143 @@
+Once you've set this user up, you should go to their permissions and set
+read-write access to all databases by clicking access for the database ``*``.
+
+There may also be some additional configuration that you'll need to in order
+for the ``multinet-server`` instance to communicate with the database. This
+might include port forwarding and establishing firewall rules, but will differ
+on a case-by-case basis.
+
+``multinet-server``
+-------------------
+
+Make sure you have python3.7 and pipenv installed. Then:
+
+1. Clone the repository: ``git clone https://github.com/multinet-app/multinet-server``.
+2. Move into the repository: ``cd multinet-server``.
+3. Copy the ``.env.default`` file: ``cp .env.default .env`` (or symlink it: ``ln
+   -s .env.default .env``).
+4. Create :ref:`OAuth2 Credentials` and make sure to copy them to your .env file.
+5. Set ALLOWED_ORIGINS to the address for your ``multinet-client`` instance,
+   including the method (e.g. https://multinet.app, yours will be different)
+6. Set your ARANGO_PASSWORD to the ``root`` password for ArangoDB, and your 
+   ARANGO_READONLY_PASSWORD to the password you created for the ``readonly`` user.
+7. Install the Python dependencies: ``pipenv install``.
+8. Run the server application in production mode with:
+   ``pipenv run gunicorn multinet.app:app``. **NOTE:** This uses gunicorn's
+   default settings, where the app is hosted on http://127.0.0.1:8000. There are
+   cases where a socket would be best (e.g. when using reverse proxies, see 
+   `below <#using-reverse-proxies>`_). In that case, run the same command with
+   an additional flag, for example: ``pipenv run gunicorn --bind unix:./socket.sock multinet.app:app``.
+   The socket will be located in the ``multinet-server`` directory.
+
+``multinet-client``
+-------------------
+
+1. Clone the repository: ``git clone https://github.com/multinet-app/multinet-client``.
+2. Move into the repository: ``cd multinet-client``.
+3. Install the project dependencies: ``yarn install``.
+4. Build the production app: ``yarn build``.
+
+At this point, you have a few options for how to proceed. The most recommended
+way to continue is to serve these files using a production webserver such as
+`nginx <https://www.nginx.com/>`_ or `apache webserver <https://httpd.apache.org/>`_.
+See the docs for these projects for details on how to do that.
+
+Handling DNS
+------------
+
+Since all of these services should be deployed onto separate servers/vms the
+easiest way to handle accessing them is through DNS. If you have a domain, you
+can set subdomains for each program in the stack. For example:
+
+* db for the ArangoDB instance
+* api for the ``multinet-server`` application
+* @ and www for the ``multinet-client`` application
+
+There may be some additional apps that you can host on the same domain. For
+example, we host MultiLink and MultiMatrix on subdomains of our main domain.
```

Comment:
I think it's a bit misleading to say that these services are supposed to be deployed onto separate servers/vms/etc. They can be, but it's not required. I also think these examples could be a bit more clear:

```suggestion
Handling DNS
------------

All of these services can be deployed onto separate servers/vms. If this is how
you want to deploy them, and you have a domain, you can assign a subdomain (through your DNS provider) for each program in the stack. For a given example domain ``domain.com``, you could organize the services as follows:

* domain.com ``multinet-client`` application
* db.domain.com for the ArangoDB instance
* api.domain.com for the ``multinet-server`` application

There may be some additional apps that you can host on the same domain. For
example, we host MultiLink and MultiMatrix on subdomains of our main domain.
```

In this example, I removed the `@` and `www` from the prefix for the client. I don't think the `www` is necessary, what is the `@` supposed to represent?

Reply:
`@` means apex in the DNS spec (e.g. multinet.app, so no subdomain)

`www` isn't absolutely necessary, but I feel like some people still type www.example.com in full. If you don't have that subdomain pointing at the IP for the client, it won't resolve.

I'll take most of your suggestions from above and add in the www

Reply:
See 6a5c862